parquet-hadoop/src/main/java/org/apache/parquet/crypto/CryptoPropertiesFactory.java

```diff
@@ -51,20 +54,32 @@ public static CryptoPropertiesFactory loadFactory(Configuration conf) throws IOE
 
     try {
       CryptoPropertiesFactory cryptoFactory = (CryptoPropertiesFactory)cryptoPropertiesFactoryClass.newInstance();
-      cryptoFactory.initialize(conf);
       return cryptoFactory;
     } catch (InstantiationException | IllegalAccessException e) {
       throw new BadConfigurationException("could not instantiate CryptoPropertiesFactory class: "
         + cryptoPropertiesFactoryClass, e);
     }
   }
 
-  public abstract void initialize(Configuration hadoopConfig);
-
-  public abstract FileEncryptionProperties getFileEncryptionProperties(
-    Configuration fileHadoopConfig, Path tempFilePath,
-    WriteContext fileWriteContext)  throws IOException;
+  /**
+   * Get FileEncryptionProperties object which is created by the implementation of this interface. Please see
+   * the unit test (TBD) for example
+   *
+   * @param fileHadoopConfig Configuration that is used to pass the needed information, e.g. KMS uri
+   * @param tempFilePath File path of the parquet file
```

Comment:
@ggershinsky, can you make comments about what to be used by this field? Our implementation doesn't need it so far.

Reply:
I'd suggest also to add a comment to the interface itself, to remind about its purpose (this helps to understand the method parameters), something like "CryptoPropertiesFactory interface enables transparent activation of Parquet encryption. It's custom implementations produce encryption and decryption properties for each Parquet file, using the information available in Parquet file writers and readers: file path, file extended schema (in writers only) - and also Hadoop configuration properties that can pass custom parameters required by a crypto factory. A factory implementation can use or ignore any of these parameters."

As for the "tempFilePath" - "File path of the parquet file being written. Can be used for AAD prefix creation, key material management, etc. Implementations must not presume the path is permanent, as the file can be moved or renamed later"

Reply:
Good point